javascript/javascript1/week4/inClassExercises.js

```diff
@@ -0,0 +1,138 @@
+        num += 1;
+        if ((num % 5 === 0) && (num % 3 === 0)) {
+            console.log('FizzBuzz');
+        } else if (num % 5 === 0) {
+            console.log('Buzz');
+        } else if (num % 3 === 0) {
+            console.log('Fizz');
+        } else {
+            console.log(num);
+        }
+    }
+}
+number(4, 12);
+
+/////////////////////////////////////////////////////////////////Build a sentiment analyzer
+
+const positiveWords = ['happy', 'awsome', 'super', 'love', 'like', 'smile'];
+const negativeWords = ['hate', 'dislike', 'boring', 'tired', 'sad', 'awful'];
+const answer = {};
+answer.positiveWords = [];
+answer.negativeWords = [];
+
+function sentimentScoreObject(sentence) {
+    let ncounter = 0;
+    let pcounter = 0;
+    for (let i = 0; i < positiveWords.length; i++) {
+        if (sentence.search(positiveWords[i]) !== -1) {
+            pcounter += 1;
+            answer.positiveWords.push(positiveWords[i]);
+        }
+    }
+    for (let i = 0; i < negativeWords.length; i++) {
+        if (sentence.search(negativeWords[i]) !== -1) {
+            ncounter += 1;
+            answer.negativeWords.push(negativeWords[i]);
+        }
+    }
+    answer.score = pcounter - ncounter;
+}
+sentimentScoreObject('I am verry happy and boring');
+console.log(answer);
+
+//console.log(sentimentScoreObject);
+/*
+{
+  score: 3,
+  positiveWords: ['happy', 'awesome', 'super'],
+  negativeWords: [],
+}
+*/
+
+
+///////////////////////////////////////////////////////////Credit card number formatter
+
+let formatted
+let seperator = 4;
+
+function formatCreditCardNumber(num) {
+    if (typeof (num) !== 'number') {
+        return 'Please Insert your card number';
+    } else {
+        num = num.toString();
+        let seperatesCounter = 0;
+        if ((num.length % seperator) === 0) {
+            seperatesCounter = Math.floor(num.length / seperator);
+        } else {
+            seperatesCounter = Math.floor(num.length / seperator) + 1;
+        }
+        formatted = '';
+        for (let i = 0; i < seperatesCounter; i++) {
+            formatted += (num.substring(0, seperator) + " ");
+            num = num.substring(seperator);
+        }
+        return formatted;
+    }
+};
+
+const formattedCreditCardObject = formatCreditCardNumber(123456789);
+console.log(formattedCreditCardObject);
+
+
+//////////////////////////////////////////////////////////////////Character frequencies
+
+const characters = [];
+
+
+function getCharacterFrequencies(words) {
+    for (let count = 1; count < words.length + 1; count++) {
+        var n = words.substring(count - 1, count);
+        var counter = 0;
+        for (let i = 1; i < words.length + 1; i++) {
+            if (n === words.substring(i - 1, i)) {
+                counter = counter + 1;
+            }
+        }
+        characters.push({
+            character: words.substring(count - 1, count),
+            count: counter,
+        });
+    }
+}
```

Comment:
`characters` should be defined _inside_ the function and be returned by it. 

Also, check what you get when you call ` getCharacterFrequencies('happy'); ` : you get the same letter (p) in it twice which is not what you want.

It would make more sense if `characters` was an object where the keys are the letters and their values are the respective counts, so that the above function call would return `{"h: 1, "a": 1, "p": 2, "y": 1}`. Try to implement it that way 🙂